### ISSUE TITLE
unmap OTU feature

### ISSUE THREAD

**Issue by zbolander** [OpenTreeOfLife/phylografter#87]
Some studies in the past have mis-mapped OTUs to distant clades (e.g. plant to grasshopper), this causes problems when uploading into the synthesis tree; a function that would allow "unmapping" OTUs, or a function that showed what OTUs did not automatically map, would be helpful

**Comment by rhr** [comment]
Sometimes a taxon automatically maps to the wrong name (e.g., if the
original input is mispelled, and the mispelling matches the wrong name
exactly).

OTUs can be manually unmapped using the web UI. Can you clarify what is
needed here?

**Comment by zbolander** [comment]
There are user error instances where OTUs were mis-mapped in the past, but
it is sometimes not obvious, especially in larger studies, which taxa were
mis-mapped. It would be great to have an option to "unmap" all taxa for a
given study to root out these problems.

**Comment by rhr** [comment]
How would unmapping an entire study root out the problems?

**Comment by josephwb** [comment]
I don't understand the resistance on this. There are many studies where
taxa have been manually mis-mapped (I've done it, only to realize it much
later, with extensive effort). On the treemachine side, if all taxa are
'mapped' (validly or not), then it is assumed kosher (i.e. TNRS is only
applied to non-mapped taxa: this might not have been communicated clearly).
This makes tracking down problems very laborious.  If I don't trust the
mapping of a particular study, it would be very useful to be able to
"un-map" taxa as described in this request.
JWB.

**Comment by rhr** [comment]
I'm just trying to understand the problem, because it seems to me like a
workflow issue on the treemachine side, not the phylografter side.

You say that for some studies you don't trust the mapping, so you want to
blow it away entirely - even though it is likely that only a few taxa are
mapped incorrectly. What makes you not trust a study's mapping to the
extent that you want to undo all that work, most of which is sound?

You say that you realize you mis-mapped something much later, with
extensive effort. How is that effort alleviated by blowing away a study's
mapping entirely?

I am just trying to understand why you would want to undo a lot of work
that is likely to be mostly correct, if it's possible to isolate the
problem OTUs and unmap them selectively.

**Comment by josephwb** [comment]
From my experience, it is not phylografter's automatic mapping that is an
issue, it is the manual mapping by users of those taxa that phylografter
cannot deal with. If there were an "un-map" option, we could let
phylografter map what it can (which, as you mention above, is sound), then
let either 1) treemachine or 2) an expert curator try to map the leftovers.
JWB.

**Comment by rhr** [comment]
I think I see what you mean. But I'm still wondering about the process of
identifying the problematic OTUs. If that can be done independently of
phylografter, then unmapping could be selective. That seems like a better
way to go.

Enabling one-click unmapping for an entire study has implications, such as,
can any curator unmap any study? In many cases, automatic mapping is
impossible (see many of Laura's examples), so I'm hesitant to provide an
easy way to undo all that hard work.

Basically, it seems that you want to use unmapping as a brute force, blunt
instrument for making things easier on the treemachine side, while I would
prefer to find a more fine-grained solution.

**Comment by zbolander** [comment]
I think this might be hard to implement, but one solution would be to show
the higher level (e.g. family, order, etc.) taxonomy for a genus & species
in phylografter, or at least have that option. The problem I encounter most
often is that a species has been erroneously placed in a distant clade, but
this is not apparent with only genus & species names. Usually, the only way
to deal with this problem now is to upload the study again.

**Comment by rhr** [comment]
This is what I don't understand - why is it not possible to identify the
problematic OTUs and fix them?

Maybe what we need is something I have been planning to implement: quick
and easy viewing of trees as they are aligned to the taxonomy. For example:

https://dl.dropboxusercontent.com/u/1939917/stree209.html

(You might have to download the file locally to see it in Chrome.)

This would help identify those problematic OTUs very quickly by visual
inspection, and facilitate rapid correcting.

**Comment by josephwb** [comment]
That would be enormously useful.
JWB.

**Comment by zbolander** [comment]
Agreed! That would be a great addition!!
